### PR TITLE
Added Course Editing to Adminitration page

### PR DESCRIPTION
**Pull request by anoverga17**
## Motivation and Context

As part of issue https://github.com/MarkUsProject/Markus/issues/5830, admins should be able to manage courses. As part of this, they should be able to edit courses via the UI. This pull request thus aims to add an admin exclusive view that allows admins to edit the attributes of a course after it has been created.

## Your Changes

**Description**:
- Added edit course route and controller method
- Added edit course view
- Added form to update course attributes
- Added update course route and controller method
- Added links to edit course view to the admin courses list
- Added check in header and menu for current_course to also check if user is on an admin route
- Refactored admin courses controller tests to prevent repeated and unnecessary code
- Added tests for edit and update changes

The changes made add the following new view:

Admin Edit

**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (internal change to codebase, without changing functionality)

## Testing

Manual testing was done through the web interface. As an admin, a course was selected and edited accordingly. The course was then checked through the UI and database to ensure it was updated properly and the correct views were being loaded. 
Rspec tests were also added to ensure the courses were being updated.

## Checklist

- [x] I have performed a self-review of my own code.
- [x] I have verified that the pre-commit.ci checks have passed. 
- [x] I have verified that the CI tests have passed. 
- [x] I have reviewed the test coverage changes reported on Coveralls. 
- [x] I have added tests for my changes. 
- [x] I have updated the Changelog.md file.

### CODE REVIEW

File: app/views/admin/courses/_form.html.erb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      <%= f.label :is_hidden, I18n.t('assignments.section_hidden.hidden'), value: 'true' %>
+      <%= f.radio_button :is_hidden, true %>
+      <%= f.label :is_hidden, I18n.t('assignments.section_hidden.visible'),  value: 'false' %>
+      <%= f.radio_button :is_hidden, false %>
+    </span>
+  </div>
+  <p>
+    <%= f.submit t(:save),
+                 data: { disable_with: t('working') } %>
+    <%= link_to t(:cancel), admin_courses_path, class: 'button' %>
```
Comment (david-yz-liu) [inline app/views/admin/courses/_form.html.erb:27]: Not a bad idea, but our other forms don't have such a button, so to be consistent you should probably remove it.

File: app/views/admin/courses/edit.html.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+<% content_for :title, "#{t(:edit)} #{Course.model_name.human}: #{@current_course.name}" %>
+<%= render partial: 'form', formats: [:html], handlers: [:erb] %>
```
Comment (david-yz-liu) [inline app/views/admin/courses/edit.html.erb:2]: You don't need the `formats` and `handlers` option here (I think by default Rails will use the same as the current file).

File: app/controllers/admin/courses_controller.rb [hunk 3]
```diff
@@ -12,5 +12,28 @@ def index
...
       end
     end
+
+    def edit
+      @course = record
+    end
+
+    def update
+      @course = record
+      update_course(@course)
```
[the remark this answers is not in the document]
Comment (anoverga17) [reply app/controllers/admin/courses_controller.rb]: Oh yeah, the flash messages appear to work! Thanks. The `respond_to :html` handles the flash messages so I don't think anything additional needs to be added.

[the remark this answers is not in the document]
Comment (anoverga17) [reply app/controllers/admin/courses_controller.rb]: After reading the documentation, the `flash_interpolation_options` would be good to configure but I will save it for #5917

**Pull request merged**